### PR TITLE
FUI - Operation details widget

### PR DESCRIPTION
**Pull request by jsorohova**

### CODE REVIEW

File: src/apim.runtime.module.ts [hunk 1]
```diff
@@ -92,9 +92,10 @@ import { TraceClick } from "./bindingHandlers/traceClick";
 import { GraphqlService } from "./services/graphqlService";
 import { ListOfApisRuntimeModule } from "./components/apis/list-of-apis/listOfApis.runtime.module";
 import { ProductListRuntimeModule } from "./components/products/product-list/productList.runtime.module";
-import { ProductApisRuntimeModule } from "./components/products/product-apis/productApis.runtime.module";
 import { OperationListRuntimeModule } from "./components/operations/operation-list/operationList.runtime.module";
 import { DetailsOfApiRuntimeModule } from "./components/apis/details-of-api/detailsOfApi.runtime.module";
+import { OperationDetailsRuntimeModule } from "./components/operations/operation-details/operationDetails.runtime.module";
+import { ProductApisRuntimeModule } from "./components/products/product-apis/productApis.runtime.module";
```
Comment (JMach1) [inline src/apim.runtime.module.ts:98]: I don't see you binding this module, am I missing something?

File: src/components/operations/operation-details/ko/runtime/type-definition-object.html [hunk 2]
```diff
@@ -66,9 +66,7 @@
 
                 <!-- /ko -->
             </div>
-            <!-- ko if: $component.examples -->
-            <div role="cell" class="text-truncate" data-bind="text: $component.examples[property.name], attr: {title: $component.examples[property.name]}"></div>
-            <!-- /ko -->
+            <div role="cell" class="text-truncate" data-bind="text: property.example, attr: {title: property.example}"></div>
```
Comment (JMach1) [inline src/components/operations/operation-details/ko/runtime/type-definition-object.html:69]: why was this removed?

File: src/apim.runtime.module.ts (hunk 1 above)
Comment (jsorohova) [reply src/apim.runtime.module.ts:98]: It is just moved from line 95.

File: src/components/operations/operation-details/ko/runtime/type-definition-object.html (hunk 2 above)
Comment (jsorohova) [reply src/components/operations/operation-details/ko/runtime/type-definition-object.html:69]: It is not removed, it is fixed. The previous logic was weird - values were taken from the big example by field name, now I am taking examples straight for the property.

File: src/apim.runtime.module.ts (hunk 1 above)
Comment (JMach1) [reply src/apim.runtime.module.ts:98]: ooh got it

File: src/components/operations/operation-details/ko/runtime/type-definition-object.html (hunk 2 above)
Comment (JMach1) [reply src/components/operations/operation-details/ko/runtime/type-definition-object.html:69]: but (from the code) if examples are not defined, we are rendering a div, while previously we were not at all, right?
I didn't check it rendered, I'm just thinking that it might look weird if there are no examples but if you rested it, it's all good

File: src/components/operations/operation-details/react/runtime/OperationDetails.tsx [hunk 3]
```diff
@@ -0,0 +1,402 @@
...
+            hostnames?.length > 0 && setHostnames(hostnames);
+        });
+        loadOperation().then(loadedValues => {
+            setOperation(loadedValues.operation);
+            setTags(loadedValues.tags);
+            setDefinitions(loadedValues.definitions);
+            setRequest(loadedValues.operation?.request);
+            setResponses(loadedValues.operation?.getMeaningfulResponses());
+        }).finally(() => {
+            setWorking(false);
```
Comment (JMach1) [inline src/components/operations/operation-details/react/runtime/OperationDetails.tsx:83]: I don't think this works as expected? what if loadApi takes longer than loadOperation?

maybe use Promise.all?

Also maybe at least some basic error handling?

File: src/components/operations/operation-details/react/runtime/OperationDetails.tsx [hunk 4]
```diff
@@ -0,0 +1,402 @@
...
+
+    const loadOperation = async (): Promise<{operation: Operation, tags: Tag[], definitions: TypeDefinition[]}> => {
+        let operation: Operation;
+        let tags: Tag[];
+        let definitions: TypeDefinition[];
+
+        try {
+            if (operationName) {
+                operation = await apiService.getOperation(`apis/${apiName}/operations/${operationName}`);
+                tags = await apiService.getOperationTags(`apis/${apiName}/operations/${operationName}`);
```
Comment (JMach1) [inline src/components/operations/operation-details/react/runtime/OperationDetails.tsx:116]: maybe make it parallel? 
and only loadDefinitions after await Promise.all

File: src/components/operations/operation-details/react/runtime/OperationDetailsGql.tsx [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+        }
+        
+        setWorking(true);
+        loadApi().then(loadedApi => setApi(loadedApi));
+        loadGatewayInfo().then(hostnames => {
+            hostnames.length > 0 && setHostnames(hostnames);
+        });
+        loadGraph()
+            .then(graph => setGraph(graph))
+            .finally(() => setWorking(false));
```
Comment (JMach1) [inline src/components/operations/operation-details/react/runtime/OperationDetailsGql.tsx:49]: same as before with Promise.all

example of my usage:

```
const configsPromise = loadCustomWidgetConfigs(this.blobStorage, this.viewManager);
const refreshConfigs = listConfigBlobs(this.blobStorage);
Promise.all([refreshConfigs, configsPromise])
      .then(([configBlobs, configsAll]) => {
....
```

File: src/components/operations/operation-details/react/runtime/TypeDefinitions.tsx [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+                                    <Body1 block className={"truncate-text"}>
+                                        <Link href={getReferenceUrl(combinationName)} title={combinationName}>{combinationName}</Link>
+                                    </Body1>
+                                );
+                            } else {
+                                children.push(<Body1 block title={combinationName}>{combinationName}</Body1>);
+                            }                                
+                        });
+                        
+                        type = <>{children}</>;
```
Comment (JMach1) [inline src/components/operations/operation-details/react/runtime/TypeDefinitions.tsx:165]: why not just children?

type = children;

**Review by JMach1 — CHANGES_REQUESTED** [review]

File: src/components/operations/operation-details/react/runtime/TypeDefinitions.tsx (hunk 6 above)
Comment (jsorohova) [reply src/components/operations/operation-details/react/runtime/TypeDefinitions.tsx:165]: Children are multiple tags and I need to wrap them into a singular one.

Comment (JMach1) [reply src/components/operations/operation-details/react/runtime/TypeDefinitions.tsx:165]: I think just changing TS type of variable "type" should do the same but just a nitpick, feel free to leave it as is
